### PR TITLE
Add hideSignUp prop to sign in

### PR DESCRIPTION
**Pull request by jordanranz**
_Issue #, if available:_
fixes: https://github.com/aws-amplify/amplify-js/issues/6045, https://github.com/aws-amplify/amplify-js/issues/6089, https://github.com/aws-amplify/amplify-js/issues/6091, https://github.com/aws-amplify/amplify-js/issues/5790, https://github.com/aws-amplify/amplify-js/issues/5594

_Description of changes:_
- Add `hideSignUp` prop to `amplify-sign-in` component

Usage:
React:
```jsx
<AmplifyAuthenticator>
      <AmplifySignIn slot="sign-in" hideSignUp />
</AmplifyAuthenticator>
```
Other frameworks:
```
<amplify-authenticator>
    <amplify-sign-in slot="sign-in" hide-sign-up></amplify-sign-in>
</amplify-authenticator>
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx [hunk 1]
```diff
@@ -304,17 +306,22 @@ export class AmplifySignIn {
...
+                  <amplify-button
+                    variant="anchor"
+                    onClick={() => this.handleAuthStateChange(AuthState.SignUp)}
+                    data-test="sign-in-create-account-link"
+                  >
+                    {I18n.get(Translations.CREATE_ACCOUNT_TEXT)}
+                  </amplify-button>
+                </span>
+              ) : (
+                <span></span>
```
Comment (sammartinez) [inline packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:321]: Would `null` not work here? Or would this break ?

Comment (sammartinez) [reply packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:321]: Answered my own question :)

Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:321]: yea it will need the element for `flex` to work

Comment (sammartinez) [reply packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:321]: yup yup that is what I realized pulling it down and checking it out :)

**Pull request merged**